### PR TITLE
Query with persistence

### PR DESCRIPTION
**Pull request by wadetandy**
This will allow persistence methods to request specific additional
includes, stats, or fields serialization on the response.  The response
object will always include any updated records, and additionally will
sideload anything that is requested.

Note that the first commit is just cleanup of request specs to simplify the rails4/5 behavior split.  Real behavior starts in later commits.

### CODE REVIEW

File: lib/graphiti/resource/interface.rb [hunk 1]
```diff
@@ -21,6 +21,7 @@ def find(params = {}, base_scope = nil)
           _find(params, base_scope)
         end
 
+        # @api private
```
Comment (richmolj) [inline lib/graphiti/resource/interface.rb:24]: 👍

File: lib/graphiti/resource_proxy.rb [hunk 2]
```diff
@@ -91,6 +91,16 @@ def save(action: :create)
           @payload.relationships
       end
       @data, success = validator.to_a
+
+      if success
+        # If the context namespace is `update` or `create`, certain
+        # adapters will cause N+1 validation calls, so lets explicitly
+        # switch to a lookup context.
```
Comment (richmolj) [inline lib/graphiti/resource_proxy.rb:98]: smart

**Review by richmolj — APPROVED** [review]
Love it 💯 🥇 🎉 

One of our most requested features implemented in a straightforward manner, cleaning up some code along the way. Just fantastic stuff here, I really like the effort put into the specs.

**Comment by wadetandy** [comment]
Ok that makes sense.  I'm adding an enhancement to Spraypaint to allow this behavior on the client side.  Should have it done today but in the meantime if you need this working, you can put the following in your form submit method:

```javascript
submitForm() {
  if (this.record.numberField == '') {
    this.record.numberField = null
  }
  // continue with record.save()
}
```

**Comment by elDub** [comment]
This is great!

**Pull request merged**